src/crawlee/_request.py

```diff
@@ -176,6 +178,20 @@ class BaseRequestData(BaseModel):
     handled_at: Annotated[datetime | None, Field(alias='handledAt')] = None
     """Timestamp when the request was handled."""
 
+    @field_validator('payload', mode='before')
+    @classmethod
+    def _payload_validation(cls, v: str | bytes | None) -> bytes | None:
+        if isinstance(v, str):
+            v = v.encode()
+        return v
+
+    @field_serializer('payload')
+    @classmethod
+    def _payload_serializer(cls, v: bytes | None) -> str | None:
+        if isinstance(v, bytes):
+            return v.decode()
+        return None
```

Comment:
Could we do this directly on the field by using Pydantic's `Annotated`? Like we do for the other fields.

Something like this:

```python
    payload: Annotated[
        HttpPayload | None,
        BeforeValidator(lambda v: v.encode() if isinstance(v, str) else v),
        PlainSerializer(lambda v: v.decode() if isinstance(v, bytes) else None),
    ] = None
    """HTTP request payload."""
```

Reply:
Updated.

The code works correctly on Apify platform